src/leaderboard/messages/data/fetch.rs

```diff
@@ -0,0 +1,20 @@
+use super::RankedElement;
+
+/// The response type for a successful `FetchByRankRequest` or
+/// `FetchByScoreRequest`.
+#[derive(Debug, PartialEq)]
+pub struct FetchResponse {
+    elements: Vec<RankedElement>,
+}
+
+impl FetchResponse {
+    /// Constructs a new `FetchResponse`.
+    pub fn new(elements: Vec<RankedElement>) -> Self {
+        Self { elements }
+    }
+
+    /// Returns the ranked elements in the response.
+    pub fn elements(&self) -> &[RankedElement] {
+        &self.elements
+    }
```

Comment:
there should be an `into_elements(self)` as well here to avoid forcing users to clone if they want owned elements.